runner/src/server/plugins/engine/pageControllers/PageControllerBase.ts

```diff
@@ -486,127 +483,153 @@ export class PageControllerBase {
-      const { num } = request.query;
-
-      // TODO:- Refactor this into a validation method
-      if (hasFilesizeError) {
-        const reformattedErrors = fileFields.map((field) => {
-          return {
-            path: field.name,
-            href: `#${field.name}`,
-            name: field.name,
-            text: "The selected file must be smaller than 5MB",
-          };
-        });
-
-        formResult.errors = Object.is(formResult.errors, null)
-          ? { titleText: "Fix the following errors" }
-          : formResult.errors;
-        formResult.errors.errorList = reformattedErrors;
-      }
-
+  async handlePostRequest(
+    request: HapiRequest,
+    h: HapiResponseToolkit,
+    mergeOptions: {
+      nullOverride?: boolean;
+      arrayMerge?: boolean;
       /**
-       * other file related errors.. assuming file fields will be on their own page. This will replace all other errors from the page if not..
+       * if you wish to modify the value just before it is added to the user's session (i.e. after validation and error parsing), use the modifyUpdate method.
+       * pass in a function, that takes in the update value. Make sure that this returns the modified value.
        */
-      if (preHandlerErrors) {
-        const reformattedErrors: any[] = [];
-        preHandlerErrors.forEach((error) => {
-          const reformatted = error;
-          const fieldMeta = fileFields.find((field) => field.id === error.name);
-
-          if (typeof reformatted.text === "string") {
-            /**
-             * if it's not a string it's probably going to be a stack trace.. don't want to show that to the user. A problem for another day.
-             */
-            reformatted.text = reformatted.text.replace(
-              /%s/,
-              fieldMeta?.label?.text.trim() ?? "the file"
-            );
-            reformattedErrors.push(reformatted);
-          }
-        });
+      modifyUpdate?: <T>(value: T) => any;
+    } = {}
+  ) {
+    const { cacheService } = request.services([]);
+    const hasFilesizeError = request.payload === null;
+    const preHandlerErrors = request.pre.errors;
+    const payload = (request.payload || {}) as FormData;
+    const formResult: any = this.validateForm(payload);
+    const state = await cacheService.getState(request);
+    const originalFilenames = (state || {}).originalFilenames || {};
+    const fileFields = this.getViewModel(formResult)
+      .components.filter((component) => component.type === "FileUploadField")
+      .map((component) => component.model);
+    const progress = state.progress || [];
+    const { num } = request.query;
+
+    // TODO:- Refactor this into a validation method
+    if (hasFilesizeError) {
+      const reformattedErrors = fileFields.map((field) => {
+        return {
+          path: field.name,
+          href: `#${field.name}`,
+          name: field.name,
+          text: "The selected file must be smaller than 5MB",
```

Comment:
nop, not configurable. `UploadService.fileSizeLimit` is just a static value `5 * 1024 * 1024;`. I forget if the intention was so that it could be configured via env var. This would be plugged into the Document Upload API that we've written for FCDO. However, if the document upload api is configured with a smaller value and returns with a 413 (I don't know if it returns any other msg), then we display "The selected file for ${fieldname} is too large"